packages/core/src/amazonq/explorer/amazonQTreeNode.ts

```diff
@@ -34,7 +34,13 @@ export class AmazonQNode implements TreeNode {
 
     public static amazonQState: AuthState
 
-    constructor() {}
+    private constructor() {
+        if (isExtensionActive(VSCODE_EXTENSION_ID.amazonq)) {
+            // 'void' instead of await, so that the command call doesn't trigger an infinite loop
+            // on constructing these instances.
+            void vscode.commands.executeCommand('aws.amazonq.refreshToolkitQState')
```

Comment:
nit: consider rename 'aws.amazonq.refreshToolkitQState' into 'aws.toolkit.refreshToolkitQState' 

Because this command is owned by toolkit and it updates states in aws toolkit. 

Also double check isExtensionActive does not throw if Q is not installed.

Reply:
verified isExtensionActive will not throw.

Will leave the command name though, because Q owns it and registers it. Toolkit only owns and registers `aws.toolkit.amazonq.refresh`